ribbon-loadbalancer/src/main/java/com/netflix/loadbalancer/BestAvailableRule.java

```diff
@@ -47,7 +47,7 @@ public Server choose(Object key) {
         Server chosen = null;
         for (Server server: serverList) {
             ServerStats serverStats = loadBalancerStats.getSingleServerStat(server);
-            if (!serverStats.isCircuitBreakerTripped(currentTime)) {
+            if (!serverStats.isCircuitBreakerTripped(currentTime) && server.isAlive() && server.isReadyToServe()) {
```

Comment:
It would be cleaner to change the above call from `getAllServers` to `getReachableServers`

Reply:
Done, thanks. That removes the need for the isAlive() check, but I guess the isReadyToServe() check is still required (assuming that's not part of an obsolete feature?)